client/jetpack-connect/index.js

```diff
@@ -19,7 +19,8 @@ export default function() {
 	const isLoggedOut = ! user.get();
 
 	page(
-		'/jetpack/connect/:type(personal|premium|pro)/:interval(yearly|monthly)?',
+		'/jetpack/connect/:type(personal|premium|pro)/:interval(yearly|monthly)?/:locale?',
```

Comment:
Can we use a whitelist of possible locales, akin to https://github.com/Automattic/wp-calypso/pull/25378/files? (Same for other `:locale?` route args below.)